crates/rolldown/src/utils/parse_to_ast.rs

```diff
@@ -42,6 +42,7 @@ pub fn parse_to_ast(
   let (source, parsed_type) = match loader {
     Loader::Js => (source, ParseType::Js),
     Loader::Json => (json_to_esm(&source)?.into(), ParseType::Js),
+    Loader::Text => (("export default \"".to_string() + &source + "\";").into(), ParseType::Js),
```

Comment:
I believe this will met problems while the content contains `"`. Could you solve this one and add related tests in `fixtures/loaders/txt/escape_quote`.

Reply:
1. add a function `text_to_esm` in `rolldown/utils/text_to_esm.rs`
2. Convert the source to Value::String in `serde::json`
3. serilized the value use serde_json and you should able to escape the quote correctly.

Not sure if this is in best performant. @IWANABETHATGUY might have some thoughts.

Reply:
I have pushed a new commit for adding a fixture and declaring a `text_to_esm` function. I looked at esbuild's implementation, esbuild seems to use `base64` encoding for text loader to realize `escape_quote`.

https://github.com/evanw/esbuild/blob/67cbf87a4909d87a902ca8c3b69ab5330defab0a/internal/bundler/bundler.go#L269-L280

Reply:
It seems that esbuilds uses codegen of itself to escape the quote. Using oxc to do this is a little bit complex. I probably stay with serde_json.

Reply:
serde_json is quite useful~